applications/ecallmgr/src/ecallmgr_fs_xml.erl

```diff
@@ -468,8 +468,21 @@ check_dtmf_type(Props) ->
         _ -> action_el(<<"start_dtmf">>)
     end.
 
+-spec build_leg_vars(kz_json:object() | kz_term:proplist()) -> kz_term:ne_binaries().
+build_leg_vars([]) -> [];
+build_leg_vars([_|_]=Prop) -> lists:foldr(fun get_channel_vars/2, [], Prop);
+build_leg_vars(JObj) -> build_leg_vars(kz_json:to_proplist(JObj)).
+
 -spec get_leg_vars(kz_json:object() | kz_term:proplist()) -> iolist().
 get_leg_vars([]) -> [];
+get_leg_vars([Binary|_]=Binaries)
+  when is_binary(Binary) ->
+    ["[^^", ?BRIDGE_CHANNEL_VAR_SEPARATOR
+    ,string:join([kz_term:to_list(V) || V <- Binaries]
```

Comment:
Did kz_binary:join/2 not work?

Reply:
@jamesaimonetti preserved the way it was already doing in the existing clause. if i change here, then would also have to change the others for consistency. i know, i'm lazy :)